lattices/src/algebra.rs

```diff
@@ -0,0 +1,428 @@
+use std::fmt::Debug;
+
+use crate::test::cartesian_power;
+
+/// Defines a monoid structure.
```

Comment:
"Defines" kinda seems like the wrong term here? Maybe "tests" would be better? I got kinda confused here because defining something should give me back something.

Reply:
I'm actually quite confused. Are there going to be traits associated with these structures as we have with Lattices?

Reply:
yes, we will add traits in the near future. Just getting the ball rolling with this first check in.

Reply:
Then just reinforcing the point Shadaj was making about naming...